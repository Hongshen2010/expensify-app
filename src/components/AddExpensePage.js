import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

const AddExpensePage = (props) => (
    <div>
        This is from the add expense component
        <h1>Add Expense</h1>
        <ExpenseForm onSubmit={(expense) => {
            props.startAddExpense(expense);
            props.history.push('/');
        }} />
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);