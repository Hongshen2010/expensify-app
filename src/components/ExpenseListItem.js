import React from 'react';
import { Link } from 'react-router-dom';
import ExpenseList from './ExpenseList';
import { connect } from 'react-redux';

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>{amount} - {createdAt}</p>
    </div>
);

const mapStateToPorps = (state) => {
    return {
        expenses: state.expenses
    };
};

export default connect(mapStateToPorps)(ExpenseListItem);