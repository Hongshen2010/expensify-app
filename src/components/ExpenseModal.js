import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import { startRemoveExpense } from '../actions/expenses';

const ExpenseModal = (props) => {
    const { description, id } = props.expense;
    return (
        <Modal
            isOpen={!!props.selectedExpense}
            contentLabel="Delete Expense?"
            onRequestClose={props.clearSelectedExpense}
            closeTimeoutMS={200}
            ariaHideApp={false}
            className="modal"
        >
            <h3 className="modal__title">Delete the following Expense?</h3>
            <h4 className="modal__body">{description}</h4>
            <button
                className="button"
                onClick={() => {
                    props.dispatch(startRemoveExpense(id));
                    props.history.push('/');
                }}
            >
                Yes, delete it.
            </button>
            <button
                className="button button--secondary"
                onClick={() => {
                    props.history.push(props.match.url);
                }}
            >
                No, keep it.
            </button>
        </Modal>
    );
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.expense.match.params.id)
});
const mapDispatchToProps = (dispatch, props) => ({
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseModal);