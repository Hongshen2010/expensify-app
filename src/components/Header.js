import React from 'react';
import { BrowserRouter, NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expense records</h1>
        <NavLink exact to="/" activeClassName="is-active">Home </NavLink>
        <NavLink to="/create" activeClassName="is-active">create </NavLink>
        <NavLink to="/help" activeClassName="is-active">help</NavLink>
    </header>
);

export default Header;