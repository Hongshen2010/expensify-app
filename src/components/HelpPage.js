import React from 'react';

const HelpPage = () => (
    <div>
        This is from the expense help component
    </div>
);

export default HelpPage;