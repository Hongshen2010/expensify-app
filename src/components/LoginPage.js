import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h2 className="box-layout__title">Expense recorder</h2>
            <p>App to manage all your expenses</p>
            <button className="button" onClick={startLogin}>Login With Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);