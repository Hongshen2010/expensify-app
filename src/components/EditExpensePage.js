import React from 'react';
import { connect } from 'react-redux';
import { startEditExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';
import ExpenseModal from './ExpenseModal';

class EditExpensePage extends React.Component {
    constructor(props) {
        super(props);
        this.state = { selectedExpense: null };
    }
    handleDeleteExpense = () => {
        this.setState({ selectedExpense: this.props.expense });
    }
    clearSelectedExpense = () => {
        this.setState({selectedExpense: null});
    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={(expense) => {
                            this.props.dispatch(startEditExpense(this.props.expense.id, expense));
                            this.props.history.push('/');
                        }}
                    />
                    <button 
                        className="button button--secondary"
                        onClick={this.handleDeleteExpense}
                    >Remove Expense</button>
                    <ExpenseModal 
                        expense={this.props}
                        dispatch={this.props.dispatch}
                        match={this.props.match}
                        history={this.props.history}
                        selectedExpense={this.state.selectedExpense}
                        clearSelectedExpense={this.clearSelectedExpense}
                    />
                </div>
            </div>
        );
    }
};
// return a single expense instead of an array
const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});
const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense))
});
export default connect(mapStateToProps)(EditExpensePage);