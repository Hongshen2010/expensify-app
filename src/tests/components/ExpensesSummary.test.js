import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Test summary component', () => {
    const wrapper = shallow(<ExpensesSummary 
                                expenseCount={1}
                                expensesTotal={114195}
                            />);
    expect(wrapper).toMatchSnapshot();
});