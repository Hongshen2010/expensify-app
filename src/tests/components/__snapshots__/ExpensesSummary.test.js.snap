// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test summary component 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $114,195.00
  </h2>
</div>
`;
